vllm/distributed/parallel_state.py

```diff
@@ -289,6 +289,9 @@ def all_reduce(self, input_: torch.Tensor) -> torch.Tensor:
         pynccl_comm = self.pynccl_comm
         if (pynccl_comm is not None and not pynccl_comm.disabled):
             pynccl_comm.all_reduce(input_)
+        elif input_.is_cpu:
+            import intel_extension_for_pytorch as ipex
+            ipex.distributed.all_reduce(input_, group=self.device_group)
```

Comment:
Just to double check: The CPU backend uses gloo runtime for distributed inference. Therefore, it basically relies on gloo for collective communications. One exception to this is all-reduce, which we use the custom implementation in IPEX for higher performance. Is my understanding correct?

Reply:
Right